src/py/flwr/common/parametersrecord.py

```diff
@@ -0,0 +1,111 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""ParametersRecord and Array."""
+
+
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional, OrderedDict
+
+
+@dataclass
+class Array:
+    """Array type.
+
+    A dataclass containing serialized data from an array-like or tensor-like object
+    along with some metadata about it.
+
+    Parameters
+    ----------
+    dtype : str
+        A string representing the data type of the serialised object (e.g. `np.float32`)
+
+    shape : List[int]
+        A list representing the shape of the unserialized array-like object. This is used
+        to deserialize the data (depending on the serialization method) or simply as a
+        metadata field.
+
+    stype : str
+        A string indicating the type of serialisation mechanism used to generate the
+        bytes in `data` from an array-like or tensor-like object.
+
+    data: bytes
+        A buffer of bytes containing the data.
+    """
+
+    dtype: str
+    shape: List[int]
+    stype: str
+    data: bytes
+
+
+@dataclass
+class ParametersRecord:
+    """Parameters record.
+
+    A dataclass storing named Arrays in order. This means that it holds entries as an
+    OrderedDict[str, Array]. ParametersRecord objects can be viewed as an equivalent to
+    PyTorch's state_dict, but holding serialised tensors instead.
+    """
+
+    keep_input: bool
+    data: OrderedDict[str, Array] = field(default_factory=OrderedDict[str, Array])
+
+    def __init__(
+        self,
+        array_dict: Optional[Dict[str, Array]] = None,
+        keep_input: bool = False,
+    ) -> None:
+        """Construct a ParametersRecord object.
+
+        Parameters
+        ----------
+        keep_input : bool (default: False)
+            A boolean indicating whether parameters should be deleted from the input
+            dictionary inmediately after adding them to the record. If True, the
```

Comment:
Is my understanding correct that this should say `False`?
```suggestion
            dictionary inmediately after adding them to the record. If False, the
```

Reply:
yes, correct with all suggested changes.